presto-rcfile/src/main/java/com/facebook/presto/rcfile/binary/BinaryColumnEncoding.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.rcfile.binary;
+
+import com.facebook.presto.rcfile.ColumnEncoding;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import io.airlift.slice.Slice;
+
+public interface BinaryColumnEncoding
+        extends ColumnEncoding
+{
+    int getValueOffset(Slice slice, int offset);
+
+    int getValueLength(Slice slice, int offset);
+
+    void decodeValueInto(Type type, BlockBuilder builder, Slice slice, int offset, int length);
```

Comment:
The way this interface is structured. It feels more natural/readable to me if `decodeColumn` always call into `decodeValueInto` after it figures out the offset and length.